frontend_app_gallery/src/component/MainPreview.jsx

```diff
@@ -2,32 +2,88 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import classnames from 'classnames'
 
-class MainPreview extends React.Component {
+export class MainPreview extends React.Component {
   constructor (props) {
     super(props)
-
+    this.imgContainer = React.createRef()
+    this.img = React.createRef()
     this.state = {
-      height: 0,
-      width: 0,
-      imageLoaded: false
+      imageLoaded: false,
+      style: {
+        maxWidth: '100%',
+        maxHeight: '100%',
+        width: '',
+        height: ''
+      }
     }
   }
 
   onLoad ({ target: img }) {
     this.setState({
-      height: img.height,
-      width: img.width,
       imageLoaded: true
     })
   }
 
-  render () {
-    const { props, state } = this
+  fixImageSize () {
+    if (this.props.rotationAngle % 180) {
```

Comment:
from my understanding, since this function is always called on component update and if props.rotationAngle is different than 0 or 180, the computing in the requestAnimatinoFrame will happen on every "compoenentDidUpdate" until rotationAngle change.

Since componentDidUpdate can be fired a lot of time (eg. especially when it has async requests related to it) I'm a bit afraid of the javascript computation amount than can be required.
Also, ref in React are attached to the DOM, not the virtual DOM. So it cannot fully use React high performance related to the virtual DOM.

Is it possible to call ifxImageSize on componendDidUpdate only when rotationAngle change ?

If not, just do some performance check on the rotation and update process. I might be wrong.

Reply:
Pm me about this if I'm not clear enough

Reply:
Yes, It is possible to call `fixImageSize` only when the rotationAngle has changed, the size has to be changed only in this case